### ISSUE TITLE
Adding "$id" to Ajv6 results in failing to validate basic `anyOf` property

### ISSUE THREAD

**Issue by michal-kurz** [rjsf-team/react-jsonschema-form#3905]
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs)
- [X] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net), [Codesandbox.io](https://codesandbox.io) or preferably a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### What theme are you using?

core

### Version

5.x

### Current Behavior

Ajv6 fails to validate the following ([playground link](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJwcmVwcm9jZXNzb3IiOnsidHlwZSI6Im5vbmUifX0sInNjaGVtYSI6eyIkaWQiOiJhYmMiLCJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJwcmVwcm9jZXNzb3IiOnsidHlwZSI6Im9iamVjdCIsImFueU9mIjpbeyJwcm9wZXJ0aWVzIjp7InR5cGUiOnsidHlwZSI6InN0cmluZyIsImVudW0iOlsibm9uZSJdLCJkZWZhdWx0Ijoibm9uZSJ9fX1dfX19LCJ1aVNjaGVtYSI6e30sInRoZW1lIjoiZGVmYXVsdCIsImxpdmVTZXR0aW5ncyI6eyJzaG93RXJyb3JMaXN0IjoidG9wIiwidmFsaWRhdGUiOmZhbHNlLCJkaXNhYmxlZCI6ZmFsc2UsIm5vSHRtbDVWYWxpZGF0ZSI6ZmFsc2UsInJlYWRvbmx5IjpmYWxzZSwib21pdEV4dHJhRGF0YSI6ZmFsc2UsImxpdmVPbWl0IjpmYWxzZSwiZXhwZXJpbWVudGFsX2RlZmF1bHRGb3JtU3RhdGVCZWhhdmlvciI6eyJhcnJheU1pbkl0ZW1zIjoicG9wdWxhdGUiLCJlbXB0eU9iamVjdEZpZWxkcyI6InBvcHVsYXRlQWxsRGVmYXVsdHMifX19)). Oddly, playground only shows validation error for the first time I click "validate" button, but in my app this results in perpetual inability to save the form.

``` js
const schema = {
  "$id": "abc",
  "type": "object",
  "properties": {
    "preprocessor": {
      "type": "object",
      "anyOf": [
        {
          "properties": {
            "type": {
              "type": "string",
              "enum": [
                "none"
              ],
              "default": "none"
            }
          }
        }
      ]
    }
  }
}

const formData = {
  "preprocessor": {
    "type": "none"
  }
}
```
>  Error: .__not_really_there__ is a required property

For some reason, this also results in (at least some) other validation errors not being reported. For example, an error for a missing required field does not show as long as this error is present ([playground link](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJwcmVwcm9jZXNzb3IiOnsidHlwZSI6Im5vbmUifX0sInNjaGVtYSI6eyIkaWQiOiJodHRwczovL2dpdGxhYi5jb20vZmVlZHlvdS9mZWVkYm90LXNjaGVtYS9zY2hlbWEvc3RlcHMvcXVlc3Rpb24uanNvbiIsInR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7InByZXByb2Nlc3NvciI6eyJ0eXBlIjoib2JqZWN0IiwidGl0bGUiOiJQcmVwcm9jZXNzb3IiLCJhbnlPZiI6W3sicHJvcGVydGllcyI6eyJ0eXBlIjp7InR5cGUiOiJzdHJpbmciLCJlbnVtIjpbIm5vbmUiXSwiZGVmYXVsdCI6Im5vbmUifX19XX0sIm1pc3NpbmciOnsidHlwZSI6InN0cmluZyJ9fSwicmVxdWlyZWQiOlsibWlzc2luZyJdfSwidWlTY2hlbWEiOnt9LCJ0aGVtZSI6ImRlZmF1bHQiLCJsaXZlU2V0dGluZ3MiOnsic2hvd0Vycm9yTGlzdCI6InRvcCIsInZhbGlkYXRlIjpmYWxzZSwiZGlzYWJsZWQiOmZhbHNlLCJub0h0bWw1VmFsaWRhdGUiOmZhbHNlLCJyZWFkb25seSI6ZmFsc2UsIm9taXRFeHRyYURhdGEiOmZhbHNlLCJsaXZlT21pdCI6ZmFsc2UsImV4cGVyaW1lbnRhbF9kZWZhdWx0Rm9ybVN0YXRlQmVoYXZpb3IiOnsiYXJyYXlNaW5JdGVtcyI6InBvcHVsYXRlIiwiZW1wdHlPYmplY3RGaWVsZHMiOiJwb3B1bGF0ZUFsbERlZmF1bHRzIn19fQ==)).  

Removing the "$id" property solves this problem. I know v6 is depracated, but I felt this shouldn't go unreported, because it seems like a very basic broken use-case. Hopefully I'm not missing something.

The bug can be circumvented by using `discriminator` with the `anyOf`

**Comment by heath-freenome** [comment]
@michal-kurz WOW, that is an interesting bug find. If you are actually using the AJV 6 validator and you feel it's important enough, we'd welcome a fix for this. I noticed that if you run raw validation in the [playground](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJwcmVwcm9jZXNzb3IiOnsidHlwZSI6Im5vbmUifX0sInNjaGVtYSI6eyIkaWQiOiJodHRwczovL2dpdGxhYi5jb20vZmVlZHlvdS9mZWVkYm90LXNjaGVtYS9zY2hlbWEvc3RlcHMvcXVlc3Rpb24uanNvbiIsInR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7InByZXByb2Nlc3NvciI6eyJ0eXBlIjoib2JqZWN0IiwidGl0bGUiOiJQcmVwcm9jZXNzb3IiLCJhbnlPZiI6W3sicHJvcGVydGllcyI6eyJ0eXBlIjp7InR5cGUiOiJzdHJpbmciLCJlbnVtIjpbIm5vbmUiXSwiZGVmYXVsdCI6Im5vbmUifX19XX0sIm1pc3NpbmciOnsidHlwZSI6InN0cmluZyJ9fSwicmVxdWlyZWQiOlsibWlzc2luZyJdfSwidWlTY2hlbWEiOnt9LCJ0aGVtZSI6ImRlZmF1bHQiLCJsaXZlU2V0dGluZ3MiOnsic2hvd0Vycm9yTGlzdCI6InRvcCIsInZhbGlkYXRlIjpmYWxzZSwiZGlzYWJsZWQiOmZhbHNlLCJub0h0bWw1VmFsaWRhdGUiOnRydWUsInJlYWRvbmx5IjpmYWxzZSwib21pdEV4dHJhRGF0YSI6ZmFsc2UsImxpdmVPbWl0IjpmYWxzZSwiZXhwZXJpbWVudGFsX2RlZmF1bHRGb3JtU3RhdGVCZWhhdmlvciI6eyJhcnJheU1pbkl0ZW1zIjoicG9wdWxhdGUiLCJlbXB0eU9iamVjdEZpZWxkcyI6InBvcHVsYXRlQWxsRGVmYXVsdHMifX19) the error is there. Something in the validator is clearly causing this error to get lost.

**Comment by michal-kurz** [comment]
@heath-freenome Yes, we are using ajv6, and we circumvented this by adding discriminators to all our `oneOf`/`anyOf` fields. So fixing this is not very important for us - especially since we would love to upgrade to ajv8 once we get the time to debug and fix the terrible performance we're getting with it.

I don't know much about how validation rjsf works, so fixing this sounds quite intimidating to me compare to what it's worth. It might be useful for someone else, but it's hard for me to justify fixing this, given it's a bug in a deprecated library I know little about, has a clear workaround, and (judging from file update history) exists for at least six months without being reported . I don't think I will find time to fix this anytime soon - unless I find myself having to familiarize myself with validators for some a different reason in the future.

**Comment by michal-kurz** [comment]
@heath-freenome Thinking about it, it might be worth asserting whether I'm right with my assumption that this bug exists for over six months. 
- If this occurs in a 6 month old version, that would suggest that fixing this is not all that important
- If it does not occur in a 6 months old version, that should mean that it broke as result of a change in `@rjsf/utils` (or less likely `ajv`), which may be worth paying closer attention to.

Is it possible to access deployed playgrounds for past versions of rjsf?

**Comment by heath-freenome** [comment]
Closing this as v6 no longer support AJV 6
